lang/cs/Org.Apache.REEF.Wake.Tests/SerializableTransportTest.cs

```diff
@@ -0,0 +1,172 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Reactive;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Org.Apache.REEF.Wake.Remote;
+using Org.Apache.REEF.Wake.Remote.Impl;
+using Org.Apache.REEF.Wake.Util;
+
+namespace Org.Apache.REEF.Wake.Tests
+{
+    [TestClass]
+    public class SerializableTransportTest
+    {
+        [TestMethod]
+        public void TestSerializableTransportServer()
+        {
+            int port = NetworkUtils.GenerateRandomPort(6000, 7000);
+
+            BlockingCollection<WritableString> queue = new BlockingCollection<WritableString>();
+            List<string> events = new List<string>();
+
+            IPEndPoint endpoint = new IPEndPoint(IPAddress.Any, port);
+            var remoteHandler = Observer.Create<TransportEvent<WritableString>>(tEvent => queue.Add(tEvent.Data));
+
+            using (var server = new SerializableTransportServer<WritableString>(endpoint, remoteHandler))
+            {
+                server.Run();
+
+                IPEndPoint remoteEndpoint = new IPEndPoint(IPAddress.Parse("127.0.0.1"), port);
+                using (var client = new SerializableTransportClient<WritableString>(remoteEndpoint))
+                {
+                    client.Send(new WritableString("Hello"));
+                    client.Send(new WritableString(", "));
+                    client.Send(new WritableString("World!"));
+
+                    events.Add(queue.Take().Data);
+                    events.Add(queue.Take().Data);
+                    events.Add(queue.Take().Data);
+                } 
+            }
+
+            Assert.AreEqual(3, events.Count);
```

Comment:
If you can take from the queue 3 times successfully, the count is 3 obviously. You might want to verify no data missing and no extra data.